test/integration/ClusterEncryptionTest.java

```diff
@@ -0,0 +1,126 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.vaticle.typedb.client.test.integration;
+
+import com.vaticle.typedb.client.TypeDB;
+import com.vaticle.typedb.client.api.TypeDBClient;
+import com.vaticle.typedb.client.api.TypeDBCredential;
+import com.vaticle.typedb.client.common.exception.TypeDBClientException;
+import com.vaticle.typedb.common.test.cluster.TypeDBClusterRunner;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_CONNECTION_CLOSED_UKNOWN;
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_SERVER_NOT_ENCRYPTED;
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_SSL_CERTIFICATE_NOT_VALIDATED;
+import static com.vaticle.typedb.client.common.exception.ErrorMessage.Client.CLUSTER_UNABLE_TO_CONNECT;
+import static org.junit.Assert.fail;
+
+public class ClusterEncryptionTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ClusterEncryptionTest.class);
+    private static final String USERNAME = "admin";
+    private static final String PASSWORD = "password";
+    private static final Path CLUSTER_CA = Path.of("test/integration/resources/dev-root-ca.pem");
+    private static final Path WRONG_CA = Path.of("test/integration/resources/wrong-root-ca.pem");
+
+    private static final TypeDBCredential ENCRYPTED_CREDENTIALS = new TypeDBCredential(USERNAME, PASSWORD, CLUSTER_CA);
+    private static final TypeDBCredential MISCONFIGURED_CREDENTIALS = new TypeDBCredential(USERNAME, PASSWORD, WRONG_CA);
+    private static final TypeDBCredential UNENCRYPTED_CREDENTIALS = new TypeDBCredential(USERNAME, PASSWORD, false);
+
+    private TypeDBClusterRunner clusterRunner;
+
+    @After
+    public void shutDownCluster() {
+        if (clusterRunner != null) {
+            clusterRunner.stop();
+            clusterRunner.deleteFiles();
+            clusterRunner = null;
+        }
+    }
+
+    private void createCluster(int nodes, Map<String, String> serverOpts) {
+        if (clusterRunner == null) shutDownCluster();
+        clusterRunner = TypeDBClusterRunner.create(Paths.get("."), nodes, serverOpts);
+    }
+
+    @Test
+    public void encrypted_server_connection_tests() {
+        createCluster(1, Collections.singletonMap("--server.encryption.enable", "true"));
```

Comment:
Do you have to tell the server to use the `CLUSTER_CA` anywhere? Right now you're using the default cluster CA? If so, you should be accessing that one instead of having a copy of it as `dev-root-ca.pem`, or you can inject the `dev-root-ca.pem` into the cluster via the configuration in `createCluster()`

Reply:
I think we should then be providing the (absolute?) path of the `dev-root-ca.pem` to use into the runner using the options the same way you're overriding the `server.encryption.enable` option